### PR TITLE
UIU-2942: Assign/unassign a users affiliations adjustments

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIU-2942](https://issues.folio.org/browse/UIU-2942) - Assign/unassign a users affiliations adjustments
## Approach

- Hide tenants `isCentral: true` and affiliations with `isPrimary: true`

https://github.com/folio-org/ui-users/assets/116072773/b3c2173b-104b-407e-b1c7-984ceae1b05a

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Comment by usavkov-epam** [comment]
Scenarios 2 and 3 haven't worked for me (https://issues.folio.org/browse/UIU-2942)


https://github.com/folio-org/ui-users/assets/88109087/d87d7bb9-aa37-4e79-9465-c2e57900f6db


https://github.com/folio-org/ui-users/assets/88109087/c6b1af74-5d3c-4d20-a226-dbd50a862da8

Does it work for you locally?

P.S. Please attach screen records with the implemented functionality.

File: src/components/AffiliationsManager/AffiliationsManager.test.js [hunk 1]
```diff
@@ -86,27 +86,21 @@ describe('AffiliationsManager', () => {
...
-        expect(await screen.findAllByRole('row')).toHaveLength((affiliations.length - 3) + 1);
+        expect(await screen.findAllByRole('row')).toHaveLength((affiliations.length - 3));
 
         await userEvent.click(await screen.findByLabelText(/Clear selected filters for/));
 
-        expect(await screen.findAllByRole('row')).toHaveLength(affiliations.length + 1);
+        expect(await screen.findAllByRole('row')).toHaveLength(affiliations.length);
       });
 
       it('should reset search and filters when \'Reset all\' button was clicked', async () => {
```
Comment (usavkov-epam) [inline src/components/AffiliationsManager/AffiliationsManager.test.js:96]: This test is deprived of semantic checkpoints after the expected results are thrown out.

**Comment by usavkov-epam** [comment]
A user, permitted to affiliate users in a tenant (in the example it's 'university'), can't select that tenant in the affiliations manager.

https://github.com/folio-org/ui-users/assets/88109087/db3b421a-8bee-4529-b853-e8416fe3e157

**Review by BogdanDenis — APPROVED** [review]

**Comment by usavkov-epam** [comment]
A user that is NOT affiliated with a tenant can assign/unassign it in the affiliation manager:

https://github.com/folio-org/ui-users/assets/88109087/4a20cab0-a27b-4d69-ba81-d8a8f92727ba

---

Please, kindly ensure that ALL requirements are met prior to requesting a re-review. [UIU-2942](https://issues.folio.org/browse/UIU-2942)

File: src/components/AffiliationsManager/AffiliationsManagerModal/AffiliationsManagerModal.js [hunk 2]
```diff
@@ -43,6 +44,8 @@ const INITIAL_FILTERS = {};
 const AffiliationManagerModal = ({ onClose, onSubmit, userId }) => {
   const [isFiltersVisible, toggleFilters] = useToggle(true);
   const [filters, dispatch] = useReducer(filtersReducer, INITIAL_FILTERS);
+  const stripes = useStripes();
+  const currentUserTenants = useMemo(() => stripes.user?.user?.tenants || [], [stripes]);
```
Comment (usavkov-epam) [inline src/components/AffiliationsManager/AffiliationsManagerModal/AffiliationsManagerModal.js:48]: redundant useMemo

**Review by usavkov-epam — CHANGES_REQUESTED** [review]
The assignment is broken.

https://github.com/folio-org/ui-users/assets/88109087/cd993adf-f1ef-4b7e-90c3-0b5fbf975b65

**Review by usavkov-epam — APPROVED** [review]
There is one still not resolved comment https://github.com/folio-org/ui-users/pull/2557/files#r1332544019

**Pull request merged**